File: apps/ios/Reels/Features/Mood/MoodModels.swift
import Foundation

// MARK: - Mood Models

enum MoodType: String, CaseIterable, Codable {
    case nostalgic = "NOSTALGIC"
    case adventurous = "ADVENTUROUS"
    case heartbroken = "HEARTBROKEN"
    case hype = "HYPE"
    case chill = "CHILL"
    case romantic = "ROMANTIC"
    case mysterious = "MYSTERIOUS"
    case inspired = "INSPIRED"
    case melancholic = "MELANCHOLIC"
    case cozy = "COZY"

    var emoji: String {
        switch self {
        case .nostalgic: return "📼"
        case .adventurous: return "🏔️"
        case .heartbroken: return "💔"
        case .hype: return "🔥"
        case .chill: return "🌊"
        case .romantic: return "💕"
        case .mysterious: return "🔮"
        case .inspired: return "✨"
        case .melancholic: return "🌧️"
        case .cozy: return "☕"
        }
    }

    var localizedKey: String {
        return "mood.\(rawValue.lowercased())"
    }
}

struct MoodSuggestion: Codable, Identifiable {
    let id: String
    let filmId: String
    let filmTitle: String
    let filmYear: Int?
    let filmPosterPath: String?
    let mood: String
    let matchExplanation: String
    let matchStrength: Double
    let source: String
}

struct MoodTwin: Codable, Identifiable {
    let userId: String
    let displayName: String?
    let image: String?
    let sharedFilmCount: Int
    let mood: String

    var id: String { userId }
}

struct SetMoodInput: Encodable {
    let mood: String
}

struct SetMoodResponse: Codable {
    let moodId: String
    let suggestions: [MoodSuggestion]
    let moodTwins: [MoodTwin]
}

struct GetSuggestionsResponse: Codable {
    let suggestions: [MoodSuggestion]
    let moodTwins: [MoodTwin]
}

struct MoodHistoryEntry: Codable, Identifiable {
    let id: String
    let mood: String
    let isActive: Bool
    let selectedAt: String
}

struct MoodHistoryResponse: Codable {
    let history: [MoodHistoryEntry]
    let currentMood: String?
}

struct TagFilmInput: Encodable {
    let filmId: String
    let mood: String
}

struct ExpressInterestInput: Encodable {
    let targetUserId: String
}

struct ExpressInterestResponse: Codable {
    let success: Bool
    let isMatch: Bool
}

File: apps/android/app/src/main/java/com/reels/app/features/mood/MoodModels.kt
package com.reels.app.features.mood

import kotlinx.serialization.Serializable

enum class MoodType(val value: String, val emoji: String) {
    NOSTALGIC("NOSTALGIC", "📼"),
    ADVENTUROUS("ADVENTUROUS", "🏔️"),
    HEARTBROKEN("HEARTBROKEN", "💔"),
    HYPE("HYPE", "🔥"),
    CHILL("CHILL", "🌊"),
    ROMANTIC("ROMANTIC", "💕"),
    MYSTERIOUS("MYSTERIOUS", "🔮"),
    INSPIRED("INSPIRED", "✨"),
    MELANCHOLIC("MELANCHOLIC", "🌧️"),
    COZY("COZY", "☕");

    companion object {
        fun fromValue(value: String) = entries.first { it.value == value }
    }
}

@Serializable
data class MoodSuggestion(
    val id: String,
    val filmId: String,
    val filmTitle: String,
    val filmYear: Int? = null,
    val filmPosterPath: String? = null,
    val mood: String,
    val matchExplanation: String,
    val matchStrength: Double,
    val source: String,
)

@Serializable
data class MoodTwin(
    val userId: String,
    val displayName: String? = null,
    val image: String? = null,
    val sharedFilmCount: Int,
    val mood: String,
)

@Serializable
data class SetMoodRequest(val mood: String)

@Serializable
data class SetMoodResponse(
    val moodId: String,
    val suggestions: List<MoodSuggestion>,
    val moodTwins: List<MoodTwin>,
)

@Serializable
data class GetSuggestionsResponse(
    val suggestions: List<MoodSuggestion>,
    val moodTwins: List<MoodTwin>,
)

@Serializable
data class MoodHistoryEntry(
    val id: String,
    val mood: String,
    val isActive: Boolean,
    val selectedAt: String,
)

@Serializable
data class MoodHistoryResponse(
    val history: List<MoodHistoryEntry>,
    val currentMood: String? = null,
)

@Serializable
data class TagFilmRequest(val filmId: String, val mood: String)

@Serializable
data class ExpressInterestRequest(val targetUserId: String)

@Serializable
data class ExpressInterestResponse(val success: Boolean, val isMatch: Boolean)
